### ISSUE TITLE
Rename NamingStrategy to FieldNamingStrategy

### ISSUE THREAD

**Issue by Chuckame** [avro-kotlin/avro4k#178]
## Current issue
Now, defining a naming strategy impacts both records' and fields' name... Well it seems. It only applies the naming strategy to field names (and the fixed names as it relies on the field name).

## Proposal
Have a clear `FieldNamingStrategy` and improve the interface:

```kotlin
public interface FieldNamingStrategy {
    public fun resolve(
        descriptor: SerialDescriptor,
        elementIndex: Int,
    ): String

    public companion object Builtins {
        public object NoOp : FieldNamingStrategy {
            override fun resolve(
                descriptor: SerialDescriptor,
                elementIndex: Int,
            ): String = descriptor.getElementName(elementIndex)
        }

        public object SnakeCase : FieldNamingStrategy {
                // ...
        }

        public object PascalCase : {
                // ...
        }
    }
}
```

**Comment by Chuckame** [comment]
Done in #182

**Comment by Chuckame** [comment]
open it back, waiting for a real release

**Comment by Chuckame** [comment]
Released in [v2.0.0](https://github.com/avro-kotlin/avro4k/releases/tag/v2.0.0)
